### ISSUE TITLE
5.17.0 is using rewrite-bom 8.11.3 and not 8.11.5

### ISSUE THREAD

**Issue by gsmet** [openrewrite/rewrite-maven-plugin#691]
As part of the `quarkus-updates` project, we have been struggling a bit to find a setup that would actually reflect the current version used by the Maven plugin and the Gradle plugin.

Typically, when developing our recipes and tests, we want to make sure that we are using the version of the rewrite-bom that is used in the Maven plugin - and ideally it would be the same one used in the Gradle plugin or we will have some discrepancies.

And similarly, for the Gradle plugin.

A good example of this issue is that a new version of the Maven plugin was released yesterday, apparently together with the core project but the Maven plugin is still referencing `8.11.3`, while the core project (with I think the fix I wanted) is at `8.11.5`.

I can enforce `8.11.5` in my project but when running the Maven plugin, the jars around will be at `8.11.3` thus not having the fix.

I was wondering if there was a good strategy to actually make sure that we are using consistent versioning, apart from manually checking each update?

Also was it a mistake that `5.17.0` of the Maven plugin is pointing to `8.11.3` and not `8.11.5`?

**Comment by timtebeek** [comment]
Hi @gsmet ; Ideally these issues are rare because of some automations we use, and I'm not quite sure how this one slipped in. I've flushed it out with a new release though, and followed up with a release of rewrite-quarkus for good measure.

To give you a bit of background: all our repositories can have individual patch releases, and depend on each other's latest version. For the Maven plugin we have a small step ahead of a release to ensure we pick up any new versions before a release:
https://github.com/openrewrite/rewrite-maven-plugin/blob/609deb5040bfcfb6f3c60bddf4f7d50009e284b5/.github/workflows/publish.yml#L30-L33

When I just ran the release for rewrite-maven-plugin it did indeed pick up the version of `<rewrite.version>8.11.5</rewrite.version>` that was released just before: https://github.com/openrewrite/rewrite-maven-plugin/commit/91600eef17d66889c33779bf683e7c45f393f13a

We (I ;)) take extra care to only release the next modules and plugins downstream once the previous ones are fully available from Maven Central. That's also why I'm puzzled how we ended up not aligning as well with the last plugin release.

Separate of any individual patch releases, we do a full release train once every two weeks or so, which sees updates to all modules and plugins, and culminates in a release of [our `rewrite-recipe-bom`](https://github.com/openrewrite/rewrite-recipe-bom) (note that's separate from `rewrite-bom`). That helps align the version numbers for the majority of our users, but the Maven or Gradle plugins aren't in there, as those versions can't be managed with a BOM as far as I know. We _could_ add the plugins to the BOM if that would help your case, with the note that it is a bit odd to have them there of course.

Open to other options or suggestions, and happy to help work through any issues that might remain.

**Comment by gsmet** [comment]
OK, I understand things better now.

Yeah, you cannot define the versions of plugins in a BOM.

I would not add the Maven and Gradle plugins as a dependency in the `rewrite-recipe-bom` but I think it would be nice to have their versions in a comment if it's doable so that we could align everything from a quick look at `rewrite-recipe-bom`.

I'm not sure how you build this BOM though and if it would be easy to do it.

Another thing that would be extremely helpful is if you were able to inject the versions of the plugins in the GitHub releases notes of the `rewrite-recipe-bom` releases. When Dependabot creates a PR (typically here: https://github.com/quarkusio/quar), the release notes are accessible so just having a look at them would give us the information and we would align the plugins right away.
Obviously, if it's just us having this problem, it's probably not worth it but I think it could maybe help others too.

I'm not exactly sure how pluggable your release process is but https://github.com/hub4j/github-api/ might come handy, we use it a lot for our automation.

**Comment by timtebeek** [comment]
Thanks for the suggestions! I like to option of adding the version numbers to the release notes; I'll see if I can fit that in. For now I've added the compatible plugin versions to the release notes of the new rewrite-recipe-bom; hope that already helps:
https://github.com/openrewrite/rewrite-recipe-bom/releases/tag/v2.5.4

**Comment by gsmet** [comment]
Yeah, I can confirm it's really useful: https://github.com/quarkusio/quarkus-updates/pull/99
